File: dotnet/Sers/Sers.Core/Sers.Core.Station/Module/Api/ApiDesc/SsModel.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Sers.Core.Module.SsApiDiscovery.ApiDesc.Attribute;

namespace Sers.Core.Module.Api.ApiDesc
{
    [JsonObject(MemberSerialization.OptIn)]
    public class SsModel
    {       


        public Func<ArraySegment<byte>, object[]> OnDeserialize;

        public object[] Deserialize(ArraySegment<byte> bytes)
        {
            return OnDeserialize(bytes);
        }






        #region JsonProperty




        /// <summary>
        /// 数据类型。 可为 object、string、int32、int64、float、double、bool、datetime 或 SsModelEntity的name
        /// </summary>
        [JsonProperty]
        [SsExample("string")]
        [SsDescription("数据类型。 可为 object、string、int32、int64、float、double、bool、datetime 或 SsModelEntity的name")]
        public string type;

        /// <summary>
        ///  数据模式。只可为 value、object、array
        /// </summary>
        [JsonProperty]
        [SsExample("value")]
        [SsDescription("数据模式。只可为 value、object、array")]
        public string mode;

        /// <summary>
        /// 描述
        /// </summary>
        [JsonProperty]
        [SsExample("")]
        [SsDescription("描述")]
        public string description;

        /// <summary>
        /// 默认值
        /// </summary>
        [JsonProperty]
        [SsExample("")]
        [SsDescription("默认值")]
        public object defaultValue;

        /// <summary>
        /// 示例值
        /// </summary>
        [JsonProperty]
        [SsExample("")]
        [SsDescription("示例值")]
        public object example;


        [JsonProperty] 
        [SsDescription("实体")]
        public List<SsModelEntity> models;

        #endregion



    }
}

File: java/Sers.ServiceStation/src/Sers/Core/Module/Api/ApiDesc/SsModel.java
package Sers.Core.Module.Api.ApiDesc;

import Sers.Core.Util.Data.ArraySegment;

public class SsModel {
	
	public interface IOnDeserialize {
		Object[] deserialize(ArraySegment bytes);
	}

	private IOnDeserialize onDeserialize;
	
	public void onDeserialize_Set(IOnDeserialize onDeserialize) {
		this.onDeserialize=onDeserialize;
	}
	
	
	 public Object[] deserialize(ArraySegment  bytes)
     {
		 return onDeserialize.deserialize(bytes);        
     }
	 
	  //region JsonProperty
		
		/**
		 *  数据类型。 可为 String、int32、int64、float、double、bool、datetime 或 SsModelEntity的name
		 */
		public String type;
		
		/** 
		 *   数据模式。只可为 value、object、array
		 */
		public String mode;
		
		/**
		 *  描述
		 */
		public String description;
		
		/**
		 *  默认值
		 */
		public Object defaultValue;
		
		/**
		 *  示例值
		 */ 
		public Object example;		
		
		 
		public SsModelEntity[] models;

      //endregion
}
